assets/js/components/settings/SettingsRenderer.js

```diff
@@ -42,8 +55,13 @@ export default function SettingsRenderer( { slug, isOpen, isEditing } ) {
 		}
 	}, [ rollbackSettings, haveSettingsChanged, isDoingSubmitChanges, isEditing ] );
 
-	if ( ! isOpen ) {
+	if ( ! isOpen || ! doesModuleExist ) {
 		return null;
+	} else if ( isOpen && ! connected ) {
+		if ( !! SettingsSetupIncompleteComponent ) {
+			return <SettingsSetupIncompleteComponent />;
+		}
+		return <DefaultSettingsSetupIncomplete slug={ slug } />;
```

Comment:
We can simplify this to 
```suggestion
		return <SettingsSetupIncompleteComponent slug={ slug } />;
```

We only need to define `moduleDefaults.SettingsSetupIncompleteComponent = DefaultSettingsSetupIncomplete` in `modules.js` (see comment below).

This is slightly different than the IB but I think we should apply defaults consistently where possible and there's no reason why we can't pass `slug` to both components.